src/tests/ftest/pool/destroy_rebuild.py

```diff
@@ -33,7 +33,18 @@ def test_destroy_while_rebuilding(self):
         """
         # Get the test parameters
         targets = self.server_managers[0].get_config_value("targets")
-        ranks = self.params.get("rank_to_kill", "/run/testparams/*")
+
+        # Select 3 ranks to rebuild
+        # rank 0 plus 1 host from access-point list, and 1 host from non-access-point list
+        ranks = [0]
+        server_hosts_list = list(set(self.hostlist_servers))
+        server_hosts_list.sort()
+        ap_hosts_list = list(set(self.access_points))
+        ap_hosts_list.sort()
+        non_ap_hosts_list = [ht for ht in server_hosts_list if ht not in ap_hosts_list]
+        ranks.append(self.server_managers[0].get_host_ranks(ap_hosts_list[-1])[0])
+        ranks.append(self.server_managers[0].get_host_ranks(non_ap_hosts_list[-1])[0])
```

Comment:
Rank information should be obtained through the `DaosServerManager` class.  Also we need to obtain the pool leader rank from the pool after it has been created.
```suggestion
        all_ranks = self.server_managers[0].ranks.keys()
        ap_ranks = self.server_managers[0].get_host_ranks(self.access_points)
        non_ap_ranks = list(set(all_ranks) - set(ap_ranks))
        
        # After creating a pool get the pool leader rank
        pool.set_query_data()
        leader_rank = pool.query_data["response"]["leader"]
        ap_ranks.remove(leader_rank)
        
        # Select the following ranks to stop:
        #  - the pool leader rank
        #  - a random rank that is not an access point
        #  - a random rank this is an access point and not the pool leader
        self.log.debug(
            "Engine ranks:  pool leader=%s, access points=%s, other=%s",
            leader_rank, ap_ranks, non_ap_ranks)
        ranks = [pool_leader_rank]
        ranks.append(random.choice(ap_ranks)  # nosec
        ranks.append(random.choice(non_ap_ranks)  # nosec
```

Reply:
For efficiency, the selection of which ranks to stop could move to after the pool is created and step #2 (which we should update to use the dmg pool query information instead of using pydaos, right @daltonbohning?) could then benefit from the already ran `dmg pool query`, e.g.
```
        # 2.
        self.log_step("Verify the pool information before starting rebuild")
        assertEqual(
            pool.query_data['response']['total_engines'], self.server_managers[0].engines,
            'Unexpected number of total pool engines')
        assertEqual(
            pool.query_data['response']['total_targets'], self.server_managers[0].engines * targets,
            'Unexpected number of total pool targets')
        assertEqual(
            pool.query_data['response']['disabled_targets'], 0,
            'Unexpected number of disabled pool targets')
```